cpp/include/cuspatial/detail/utility/offset_to_keys.cuh

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <thrust/binary_search.h>
+#include <thrust/distance.h>
+
+/** @brief Given list offset and row `i`, return a unique key that represent the list of `i`.
```

Comment:
The only difference between this functor and `upper_bound` is that it returns an index rather than an iterator. So just use the same style of docs.

```suggestion
/** @brief Returns the index of the first element in the range `[first, last)` such that `value < element` is true (i.e. strictly greater), or `distance(first, last)` if no such element is found.
```

And update the parameters names to match...